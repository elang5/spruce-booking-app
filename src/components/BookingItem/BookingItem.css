.booking-item {
  background: white;
  font-size: 12px;
}

.address {
  display: flex;
  flex-flow: column wrap;
}

.street {
  margin-bottom: 4px;
}

.booking-item.booking-time {
  text-align: right;
  padding-right: 10px;
}