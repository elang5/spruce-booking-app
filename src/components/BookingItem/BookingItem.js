import React from 'react'
import BookingSpan from '../BookingSpan/BookingSpan'

export default function BookingListItem({ booking }) {
  return (
    <li className="booking-item">
      <BookingSpan 
        className='customer'
        type={booking.name} />
      <BookingSpan 
        className='email'
        type={booking.email} />
      <BookingSpan 
        className='address' 
        type={`${booking.address} ${booking.city}, ${booking.state}, ${booking.zip}`} />
      <BookingSpan 
        className='booking-type'
        type={booking.bookingtype} />
      <BookingSpan 
        className='booking-time'
        type={booking.datetime} />
    </li>
  )
}
