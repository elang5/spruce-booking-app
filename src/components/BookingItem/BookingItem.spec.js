import React from "react";
import ReactDOM from "react-dom";
import BookingItem from "./BookingItem";
import renderer from "react-test-renderer";

describe("BookingItem Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(< BookingItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(<BookingItem/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
