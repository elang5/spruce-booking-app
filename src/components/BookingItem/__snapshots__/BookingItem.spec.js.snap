// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingItem Component renders the UI as expected 1`] = `
Array [
  <div
    className="booking-item cell customer"
  >
    
  </div>,
  <div
    className="booking-item cell email"
  >
    
  </div>,
  <div
    className="booking-item cell address"
  >
    <div
      className="street"
    >
      
    </div>
    <div
      className="city-state-zip"
    >
      , , 
    </div>
  </div>,
  <div
    className="booking-item cell booking-type"
  >
    Dog Walk
  </div>,
  <div
    className="booking-item cell booking-time"
  >
    Invalid Date
  </div>,
]
`;
